docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,871 @@
+    └── 99_role-cloud-creds-secret-reader.yaml
+
+2 directories, 38 files
+```
+
+### Remove Machines and MachineSets
+
+Remove the control-plane Machines and compute MachineSets, because we'll be providing those ourselves and don't want to involve the
+[machine-API operator][mao]:
+
+[mao]: https://github.com/openshift/machine-api-operator
+
+```sh
+$ rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+```
+You are free to leave the compute MachineSets in if you want to create compute machines via the machine API, but if you do you may need to update the various references (`subnet`, etc.) to match your environment.
+
+### Make control-plane nodes unschedulable
+
+Currently [emptying the compute pools][empty-compute-pools] makes control-plane nodes schedulable. But due to a [Kubernetes limitation][kubebug], router pods running on control-plane nodes will not be reachable by the ingress load balancer. Update the scheduler configuration to keep router pods and other workloads off the control-plane nodes:
+
+[empty-compute-pools]: #empty-compute-pools
+[kubebug]: https://github.com/kubernetes/kubernetes/issues/65618
+
+```sh
+$ python3 -c '
+import yaml;
+path = "manifests/cluster-scheduler-02-config.yml"
+data = yaml.safe_load(open(path));
+data["spec"]["mastersSchedulable"] = False;
+open(path, "w").write(yaml.dump(data, default_flow_style=False))'
+```
+
+
+## Ignition Config
+
+Next, we will turn these manifests into [Ignition][ignition] files. These will be used to configure the Nova servers on boot (Ignition performs a similar function as cloud-init).
+
+[ignition]: https://coreos.com/ignition/docs/latest/
+
+```sh
+$ openshift-install create ignition-configs
+$ tree
+.
+├── auth
+│   ├── kubeadmin-password
+│   └── kubeconfig
+├── bootstrap.ign
+├── master.ign
+├── metadata.json
+└── worker.ign
+```
+
+
+### Infra ID
+
+The OpenShift cluster has been assigned an identifier in the form of `<cluster name>-<random string>`. You do not need this for anything, but it is a good idea to keep it around.
+You can see the various metadata about your future cluster in `metadata.json`.
+
+The Infra ID is under the `infraID` key:
+
+```sh
+$ export INFRA_ID=$(jq -r .infraID metadata.json)
+$ echo $INFRA_ID
+openshift-qlvwv
+```
+
+We'll use the `infraID` as the prefix for all the OpenStack resources we'll create. That way, you'll be able to have multiple deployments in the same OpenStack project without name conflicts.
+
+Make sure your shell session has the `$INFRA_ID` environment variable set when you run the commands later in this document.
+
+
+### Update Bootstrap Ignition
+
+There is some bootstrap configuration we weed to add explicitly to the ignition file. The IPI installer does this automatically via Terraform, but the changes do not show up in the files generated by `create ignition-configs`. This section should go away in the future.
```

Comment:
It is to the end-users: someone reading this might wonder: why am I manually making these ignition changes here? They're not necessary for the AWS UPI and they're done automatically with the OpenStack IPI. So what gives?

Maybe I could word this better?

Reply:
OK I am fine with addressing this nitpick at a later stage (possibly, when we actually get rid of what we promised we get rid of :grin: )